test/common/README.md

```diff
@@ -299,6 +299,14 @@ If `fn` is not provided, an empty function will be used.
 Returns a function that triggers an `AssertionError` if it is invoked. `msg` is
 used as the error message for the `AssertionError`.
 
+### `mustNotMutate(object)`
```

Comment:
```suggestion
### `mustNotMutate([object])`
```
Should we accept `{}` as the default input for `object`?

Reply:
I considered that, but this would at least break a case when we knowingly pass `undefined` (for example, in forEach along with a ten of various objects).

Reply:
I don't think this should account for primitive types like `undefined` because as you said, those are immutable already and there's no need to test that. `common.mustNotMutate([object])` where `object` defaults to `{}` is fine for now IMO.

Reply:
I can't agree with that.
This function should always transparently and unconditionally return any immutable value as is. Substituting `undefined` with `{}` doesn't seem to be justified.
A good example for cases where we might strongly need it is [test-fs-write-optional-params.js](https://github.com/LiviaMedeiros/node/blob/48a8bd155eebe3510f581db95d47c553d4b22add/test/parallel/test-fs-write-optional-params.js)
- To avoid code duplication, `common.mustNotMutate()` should be used inside of `testValidCb()` and `testInvalidCb()`
- We do want to pass absolutely whatever options into `testInvalid()`, and we never want it to be mutated
- We do want to allow `null` and/or `undefined` (in this particular case it's only `null`, and `undefined` explicitly tested as invalid) as valid value and test it along with other options

Additionally, writing `{}` explicitly removes all ambiguity in what will be passed, e.g. `{}` or `Object.create(null)`.

However, I agree that it doesn't look clear in the docs. How about marking it explicitly? Like this:

### `mustNotMutate([target])`
* `target` [\<any\>](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Data_types) default = `undefined`

Reply:
fair point :+1: